### DISCUSSION TITLE
Contract Interaction inside metamask snap.

### DISCUSSION THREAD

**Discussion by 1337-ishaan** [MetaMask/snaps#2056] [General]
I have a use case where user interacts with a contract, is it possible that I get the required parameters from the dapp and pass them in a request inside the snap, and let snap interact with the contract?

**Comment by 1337-ishaan** [comment]
let me know if you need further clarities on the question!

**Comment by Montoya** [comment]
Right now the Snap cannot initiate transactions so it isn't possible for a Snap to make write requests. A Snap can only read from a contract. Are you looking to read or write?

**Reply by 1337-ishaan** [reply]
I want the snap to interact with a contract, would appreciate your advices.

The other option that i currently have is setting up a nodejs server, create APIs and then call those APIs inside the sandbox, what do you think?

**Reply by Montoya** [reply]
Could you tell me more about what the contract does?

**Reply by 1337-ishaan** [reply]
I am building a snap for ZetaChain & I want to enable send & receive through the snap itself.

Pls find the contract withdraw & deposit architecture below:

More Info: [https://www.zetachain.com/docs/developers/tokens/](https://www.zetachain.com/docs/developers/tokens/)

<img width="683" alt="Screenshot 2023-12-22 at 12 59 07 AM" src="https://github.com/MetaMask/snaps/assets/64636222/50c4d8fe-ef5a-4ed5-910c-eec6326ec076">

**Reply by Montoya** [reply]
Which addresses are you using for Zetachain? Are you deriving with BIP-32 or BIP-44?

**Reply by 1337-ishaan** [reply]
I had planned on deriving both of them, because there should be transactions taking place in bitcoin and altcoins.

Would like to have your suggestion for the same.
